contracts/contract_caller.go

```diff
@@ -0,0 +1,67 @@
+package contracts
+
+import (
+	"context"
+	"math/big"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+type RPCClient interface {
```

Comment:
i think you added it. there is no need for it. 
SimulatedBackend already implements required interface. Please use it in tests.
For normal workflow use regular client. Not our proxy - just regular client.
You can instantiate one using `ethclient.NewClient` either with local light client or remote client.